pkg/scheduler/eventhandlers.go

```diff
@@ -118,14 +118,14 @@ func (sched *Scheduler) updateNodeInCache(oldObj, newObj interface{}) {
 		klog.Errorf("scheduler cache UpdateNode failed: %v", err)
 	}
 
-	// Only activate unschedulable pods if the node became more schedulable.
 	// We skip the node property comparison when there is no unschedulable pods in the queue
-	// to save processing cycles. We still trigger a move to active queue to cover the case
-	// that a pod being processed by the scheduler is determined unschedulable. We want this
-	// pod to be reevaluated when a change in the cluster happens.
+	// to save processing cycles.
 	if sched.SchedulingQueue.NumUnschedulablePods() == 0 {
-		sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(queue.Unknown)
```

Comment:
Can you confirm the below logic is covered somewhere if remove the move request here?
> We still trigger a move to active queue to cover the case that a pod being processed by the scheduler is determined unschedulable. We want this pod to be reevaluated when a change in the cluster happens.

The only difference I can see is that `moveRequestCycle` is incremented which might push the pod into `podBackoffQ` instead of `unschedulableQ`.

Reply:
I don't understand why we had this anyways. The queue is empty, so nothing gets moved. Unless, of course, a new pod comes in between the nanoseconds between these 2 function calls. But that's not much different from a Pod coming in after `updateNodeInCache` finishes.

Reply:
@chendave The old logic isn't quite aligned with the comment:

> We want this pod to be reevaluated when a change in the cluster happens.

Actually if the pod failed the scheduling when this method is called, it will be moved to unschedulableQ/backoffQ instead of activeQ/backoffQ.

---

> The only difference I can see is that `moveRequestCycle` is incremented which might push the pod into `podBackoffQ` instead of `unschedulableQ`.

This is a valid point. In the old logic, the unschedulable Pod will be moved to backoffQ (moveCycle >= schedulingCycle). This is based on the old pattern to move all pods anyways. However, we will be transitioning to more fine-grained re-queuing semantics, which only respects the fact that if the event can make some pods schedulable, which has several variants:

- there are 0 unschedulable pods
- the event is a dummy no-op (like the above method, node properties is intact)
- the event doesn't make any existing unschedulable pod schedulable

In this way, we have a coherent semantics to refresh the moveCycle or not. And I believe the new logic will be more robust and easy to maintain (maybe some behavior is different with old logic, but that's fine).

(the old behavior is inconsistent and ambiguous - if node properties remains unchanged, the logic returns directly; if there are 0 unschedulabel pods, it continues to reset the moveCycle.)

Reply:
Let's discuss this in sig-scheduling slack. Ping me when you are online.

Reply:
Discussed with @alculquicondor offline, another option is to run `nodeSchedulingPropertiesChange` unconditionally.

The slight difference with old behavior is: the new behavior increments moveRequestCycle when node properties changes, no matter there is unschedulable pods or not; while in old logic, moveRequestCycle can be incremented when node properties don't change but no unschedulable pods are present.